core/clusters/clusterconfigs.go

```diff
@@ -169,12 +174,12 @@ func readConfig(name string, res *ClusterConfig, failOnMissing bool, restconfig
 	}
 	if err == nil && cmap != nil {
 		// Kube will give us an empty configmap if the named one does not exist,
-		// so we test for a Name to see if we foud it
+		// so we test for a Name to see if we found it
 		found = cmap.Name != ""
 		for n, v := range cmap.Data {
 			err = process(res, strings.Trim(n, "\n"), strings.Trim(v, "\n"), name)
 			if err != nil {
-				break
+				return found, err
```

Comment:
same here, philosophy :) The approach in this code is include an early return only when necessary.
If a break will cause a fall through to the return with no additional processing, then a break should be used.